docs/latest/design/egctl.md

```diff
@@ -0,0 +1,64 @@
+# Introduce egctl
+
+## Motivation
+
+EG should provide a command line tool with following capabilities:
+
+- One-step installation/uninstallation experience
+- Collect configuration from data plane
+- Update admin options such as changing log level
+- Analyse system configuration to diagnose any issues in envoy-gateway
+
+This tool is named `egctl`.
+
+## Syntax
+
+Use the following syntax to run `egctl` commands from your terminal window:
+
+```console
+egctl [command] [subcommand] [name] [flags]
+```
+
+where `command`, `name`, and `flags` are:
+
+* `command`: Specifies the operation that you want to perform on one or more resources,
+  for example `install`, `uninstall`, `proxy-config`, `version`.
+
+* `name`: Specifies the name of the specified pod. 
+
+* `flags`: Specifies optional flags. For example, you can use the `-f` or `--file` flags to specify the values for installing.
+
+If you need help, run `kubectl help` from the terminal window.
+
+## Operation
+
+The following table includes short descriptions and the general syntax for all the `egctl` operations:
+
+| Operation      | Syntax                          | Description                                               |
+| -------------- |---------------------------------|-----------------------------------------------------------|
+| `version`      | `egctl version`                 | Prints out build version information.                     |
+| `install`      | `egctl install -f FILENAME`     | Install or reconfigure EG on a cluster.                   |
+| `uninstall`    | `egctl uninstall`               | Uninstall EG from a cluster                               |
+| `proxy-config` | `egctl proxy-config SUBCOMMAND` | Retrieve information about proxy configuration from envoy |
+| `analyze`      | `egctl analyze`                 | Analyze configuration and print validation messages       |
+
+## Examples
+
+Use the following set of examples to help you familiarize yourself with running the commonly used `egctl` operations:
+
+```console
+# Install EG using the definition in custom-values.yaml
+egctl install -f custom-values.yaml
+
+# Uninstall EG
+egctl uninstall
+
+# Retrieve all information about proxy configuration from envoy
+egctl proxy-config all <pod_name>
```

Comment:
I prefer `config` over the `proxy-config` noun.

To pull config from the managed proxies:
```
egctl get config --data-plane
```

To pull config from EG:
```
egctl get config --control-plane
```

When the `--control-plane` or `--data-plane` are not specified, egctl gets config from control and data planes.

Reply:
Inspired by `istioctl`, `proxy-config` can filter xds by specific type(`endpoint`, `ecds`, `cluster` etc.)

Also, I think we don't have a debug endpoint in `control-plane`.

Reply:
@zirain you could achieve the same by `egctl config envoy-proxy cluster` . I prefer this over `egctl envoy-proxy-config cluster` because former maintains a more consistent hierarchy starting from `verb` user is trying to perform

Reply:
what about `control-plane` with `egctl config` in your mind, `ectl config control-plane`?

Reply:
I prefer `envoy-gateway` and `envoy-proxy` over `control-plane` and `data-plane` but this naming is minor for me.
we can have
```
egctl config envoy-proxy
egctl config envoy-gateway
egctl config envoy-rate-limit-service
egctl config all
```
and we can start with `egctl config envoy-proxy`

Reply:
if you want the `config` subcommand to only display config, then we can create another subcommand called `collect` which can be similar to `bug-report`

Reply:
I have no strong option on naming, will update the doc later.

Reply:
updated, PTAL